docs/features/deconstruction.md

```diff
@@ -80,23 +81,16 @@ evaluate side-effect on the left-hand-side variables
 evaluate Deconstruct passing the references directly in
 ```
 
-Note that the feature is built around the `Deconstruct` mechanism for deconstructing types.
-`ValueTuple` and `System.Tuple` will rely on that same mechanism, except that the compiler may need to synthesize the proper `Deconstruct` methods.
-
+In the case where the expression on the right is a tuple, the evaluation order becomes:
+```
+evaluate side-effect on the left-hand-side variables
+evaluate the right-hand-side and do a tuple conversion
+assign element-wise from the right to the left
```

Comment:
I basically make a fake tuple type that looks like what is on the LHS. I'm not sure what is a good way to explain that in the spec though.